lib/vtls/openssl.c

```diff
@@ -2356,9 +2356,25 @@ static CURLcode get_cert_chain(struct connectdata *conn,
     if(!pubkey)
       infof(data, "   Unable to load public key\n");
     else {
-      switch(pubkey->type) {
+      int pktype;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+      pktype = EVP_PKEY_id(pubkey);
+#else
+      pktype = pubkey->type;
+#endif
+      switch(pktype) {
       case EVP_PKEY_RSA:
-        BIO_printf(mem, "%d", BN_num_bits(pubkey->pkey.rsa->n));
+      {
+        RSA *rsa;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+        rsa = EVP_PKEY_get0_RSA(pubkey);
+        BN_print(mem, rsa->n);
```

Comment:
@vszakats Can you please give it a run on OpenSSL 1.1.0 using the certinfo.c example and see what shows for 'RSA Public Key:' when you connect to a website like ssllabs.com or something. It should show the number of bits, like
~~~
RSA Public Key:2048
rsa(n):a whole lot of hex digits should be here, but not on on the line above.
~~~